bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -513,13 +509,15 @@ void ReplicaImp::onMessage<ClientRequestMsg>(std::unique_ptr<ClientRequestMsg> m
           }
           requestsOfNonPrimary[msg->requestSeqNum()] = std::make_pair(getMonotonicTime(), msg);
```

Comment:
Please review this line again. you are actually moving ownership but still deleting the msg with the smart pointer. We will have a double free which is a security issue and undefined behavior (possible crash), won't we?

Reply:
This memory is released in line 513. Shall I change it to m.get()